packages/daemon/src/service/pipeline.ts

```diff
@@ -48,6 +48,200 @@ function execAsync(cmd: string, opts: ExecOptions): Promise<string> {
   });
 }
 
+interface RunnerOptions {
+  job: JobEntity;
+  pipeline: PipelineEntity;
+  plugins: Partial<Record<PluginTypeI, PluginInfo>>;
+  tracer: Tracer;
+  runnable: PluginRunnable;
+  datasetRoot: string;
+}
+
+interface ModelResult {
+  model: RunnableResponse;
+  modelPlugin: PluginPackage;
+}
+
+interface RunnerResult {
+  output: any;
+  dataset: any;
+  modelPath: string;
+  modelPlugin: PluginPackage;
+  dataProcess: PluginPackage;
+  datasetProcess: PluginPackage;
+}
+
+/**
+ * job runner
+ */
+export class JobRunner {
+  opts: RunnerOptions;
+  constructor(
+    opts: RunnerOptions) {
+    this.opts = opts;
+  }
+  getParams(params: string | null, ...extra: object[]): object {
+    if (params == null) {
+      return Object.assign({}, ...extra);
+    } else {
+      return Object.assign(JSON.parse(params), ...extra);
+    }
+  }
+  verifyPlugin (name: string): void {
```

Comment:
```suggestion
  assertPlugin (name: string): void {
```
"verify" does mean a bool would be returned.